### PR TITLE
UPSTREAM: 30626: prevent RC hotloop on denied pods

### PR DESCRIPTION
**Pull request by deads2k**
Fixes https://github.com/openshift/origin/issues/10407

We'll need to review similar controllers after this one, but this is our biggest risk and I'd like to see the pattern merged before pushing more upstream.

@kargakis ptal

### CODE REVIEW

File: vendor/k8s.io/kubernetes/pkg/controller/replication/replication_controller.go [hunk 1]
```diff
@@ -485,55 +494,79 @@ func (rm *ReplicationManager) manageReplicas(filteredPods []*api.Pod, rc *api.Re
...
-		// Note that if the labels on a pod/rc change in a way that the pod gets
-		// orphaned, the rs will only wake up after the expectations have
-		// expired even if other pods are deleted.
-		deletedPodKeys := []string{}
-		for i := 0; i < diff; i++ {
-			deletedPodKeys = append(deletedPodKeys, controller.PodKey(filteredPods[i]))
+
+		select {
+		case err := <-errCh:
+			// all errors have been reported before and they're likely to be the same, so we'll only return the first one we hit.
```
Comment (kargakis) [inline vendor/k8s.io/kubernetes/pkg/controller/replication/replication_controller.go]: This is ok for now but a follow-up issue to actually verify this (or at the very least that we got `diff` num of errors, meaning all goroutines failed to create) would be nice.

**Comment by kargakis** [comment]
Changes LGTM, the RS controller will need the same fix most probably though

**Comment by deads2k** [comment]
[merge]

**Comment by deads2k** [comment]
> Changes LGTM, the RS controller will need the same fix most probably though

My initial check shows that the RS controller has managed to have a different set of bugs.

**Comment by ncdc** [comment]
@deads2k will this fix #6390 either partially or fully? That will happen if you don't have MatchNodeSelector scheduler predicate enabled.

**Comment by deads2k** [comment]
> @deads2k will this fix #6390 either partially or fully? That will happen if you don't have MatchNodeSelector scheduler predicate enabled.

This won't address that one way or the other.

**Pull request merged**